addons/point_of_sale/static/src/js/screens.js

```diff
@@ -1182,7 +1182,7 @@ var ClientListScreenWidget = ScreenWidget.extend({
         
         var fields = {};
         this.$('.client-details-contents .detail').each(function(idx,el){
-            fields[el.name] = el.value;
+            fields[el.name] = el.value || false;
```

Comment:
You're right, the code is not only used to create the records, it's also used to update existing records. Forget I said anything, your code looks good :).